Sources/MapboxNavigation/CarPlayNavigationViewController.swift

```diff
@@ -141,15 +146,30 @@ open class CarPlayNavigationViewController: UIViewController, BuildingHighlighti
     }
     
     /**
-     Updates `CPMapTemplate.tripEstimateStyle` based on provided `UIUserInterfaceStyle` value.
-     
-     - parameter userInterfaceStyle: `UIUserInterfaceStyle`, which is currently used on CarPlay.
+     Updates `CPMapTemplate.tripEstimateStyle` and `CPMapTemplate.guidanceBackgroundColor`.
      */
-    func updateTripEstimateStyle(_ userInterfaceStyle: UIUserInterfaceStyle) {
-        switch traitCollection.userInterfaceStyle {
+    func updateMapTemplateStyle() {
+        var currentUserInterfaceStyle = traitCollection.userInterfaceStyle
+        // Regardless of the user interface style that was set in CarPlay settings style that is
+        // currently used in `StyleManager` will have precedence. Precedence of the style over
+        // trait collection is required for custom cases (e.g. when user is driving through the tunnel).
+        if usesNightStyleWhileInTunnel,
+           isTraversingTunnel,
+           let styleType = styleManager?.currentStyleType {
+            switch styleType {
+            case .day:
+                currentUserInterfaceStyle = .light
+            case .night:
+                currentUserInterfaceStyle = .dark
+            }
+        }
+        
+        switch currentUserInterfaceStyle {
         case .dark:
+            mapTemplate.guidanceBackgroundColor = .black
```

Comment:
It world fix the issue why the guidanceBackgroundColor doesn't change style when the bottom trip estimate banner changes.

Reply:
Yes, interestingly we cannot set any color there. For now this value is hardcoded, but in future we can consider exposing the ability to change this color by the end user.

Reply:
> in future we can consider exposing the ability to change this color by the end user.

Agreed: #3916.